src/monty/io.py

```diff
@@ -67,26 +114,32 @@ def reverse_readfile(filename: Union[str, Path]) -> Generator[str, str, None]:
     Yields:
         Lines from the file in reverse order.
     """
+    # Generate line ending
+    l_end = _get_line_ending(filename)
+
     try:
         with zopen(filename, "rb") as file:
             if isinstance(file, (gzip.GzipFile, bz2.BZ2File)):
                 for line in reversed(file.readlines()):
-                    yield line.decode("utf-8").rstrip(os.linesep)
+                    yield line.decode("utf-8").rstrip(l_end)
+
             else:
                 filemap = mmap.mmap(file.fileno(), 0, access=mmap.ACCESS_READ)
                 n = len(filemap)
                 while n > 0:
-                    i = filemap.rfind(os.linesep.encode(), 0, n)
-                    yield filemap[i + 1 : n].decode("utf-8").rstrip(os.linesep)
+                    i = filemap.rfind(l_end.encode(), 0, n)
+                    yield filemap[i + 1 : n].decode("utf-8")
```

Comment:
One more question: should we append the line ending when read reversely? (might need to look at how `readline` handles this)

In other word, should we yield the line with or without line ending?